os/board/common/crashdump.c

```diff
@@ -167,6 +311,35 @@ static int ramdump_via_uart(void)
  ****************************************************************************/
 
 #if defined(CONFIG_BOARD_CRASHDUMP)
+void crashdump_init(void)
+{
+	int index;
+
+	index = mtd_getpartitionindex("coredump");
+	if (index < 0) {
+		lldbg("partition index is wrong for coredump part name\n");
+		return;
```

Comment:
Wouldn't it be good to return some error number instead of simple return?
We can also validate crashdump_init after calling it.

Reply:
validation after crashdump_init call will not be used for any case, so its void function.
The values of partition size and name are used/validated during dump in different thread.